config/application.rb

```diff
@@ -1,6 +1,25 @@
 require_relative 'boot'
 
-require 'rails/all'
+require "rails"
+[
+  "active_record/railtie",
+  "active_storage/engine",
+  "action_controller/railtie",
+  "action_view/railtie",
+  "action_mailer/railtie",
+  "active_job/railtie",
+  #"action_cable/engine", # Enable this when installing StimulusReflex
+  "action_mailbox/engine",
+  "action_text/engine",
+  "rails/test_unit/railtie",
+  "sprockets/railtie" # Disable this after migrating to Webpacker
+].each do |railtie|
+  begin
+    require railtie
+  rescue LoadError
+  end
+end
```

Comment:
I'm quite afraid of the unwanted side effects this could have in future upgrades. It's very easy to forget to add necessary requires and waste hours trying to find out why things are broken.

Reply:
I see your point. I'd keep require rails/all for simplicity even if there's more code loaded.
But I am also ok with this solution.
We need a decision.

Reply:
ActionMailbox eats roughly 25MB of memory if it's running. Which doesn't sound like much until you realise that's _per worker_, so for UK (5 workers) that would be ~125MB of memory wasted for no reason. We definitely don't want that on all instances, right? Which means one way or another we need to avoid using `rails/all`.

It's now common practice to whitelist the Rails modules you're actually using (since ActionMailbox was added, hahaha) and having a single place where that list is explicitly declared seems pretty reasonable.